### PR TITLE
Add anchor to each example for external linking

### PR DESCRIPTION
**Pull request by CharlesGueunet**
Dear Julien,
This PR improve the gallery of TTK se we can link to https://topology-tool-kit.github.io/gallery.html#MorseMolecule for example to directly jump to the corresponding image.

Best,
Charles

### CODE REVIEW

File: gallery.html [hunk 1]
```diff
@@ -169,9 +170,8 @@ <h3>Scalar data</h3>
                   </td>
                 </tr>
                 <tr>
-                  
                   <td >
-                    <h4>Dragon demo</h4>
+
```
Comment (julien-tierny) [inline gallery.html]: I believe you shifted all the titles by one. 
thanks for fixing that!

**Review by julien-tierny — COMMENTED** [review]
hi charles, 
thanks for your PR.
it seems that it needs some fixing (you shifted all titles by one)

File: gallery.html (hunk 1 above)
Comment (julien-tierny) [reply gallery.html]: basically the title for the "dragon demo" should be here and not below

File: gallery.html [hunk 2]
```diff
@@ -190,9 +190,8 @@ <h4>Dragon demo</h4>
                 <tr height="30px" />
 
                 <tr>
-                  
                   <td >
-                    <h4>Morse persistence demo</h4>
+                    <h4><a id="DragonDemo">Dragon demo</a></h4>
```
Comment (julien-tierny) [inline gallery.html]: as you can read, the list "features in action" includes the morse-smale complex, which is not in the dragon demo, but in the next one (morsePersistence).
thanks for fixing that.

File: gallery.html (hunk 1 above)
Comment (vollia) [reply gallery.html]: oh ! right ^^ I will fix that.

**Comment by vollia** [comment]
I added some more indent fix and I have re-created the title and anchor so they are not shifted anymore. I had a bug in my macro, sorry.

**Comment by julien-tierny** [comment]
alright, looks good, thanks!

**Pull request merged**